File: AlgorithmDataStructure/LeetCode/1-TwoSum/sum.c

#include <stdio.h>

int *twoSum(int *nums, int numsSize, int target, int *returnSize){
    int *indexs = (int *)malloc(sizeof(int) * 2);
    for (int i = 0; i < numsSize; i++) {
        for (int j = i + 1; j < numsSize; j++) {
            if (nums[i] + nums[j] == target) {
                indexs[0] = i;
                indexs[1] = j;
                *returnSize = 2;
                return indexs;
            }
        }
    }
    *returnSize = 0;
    return indexs;
}


int main() {
    
    int size = 0;
    int array[4] = {2,7,11,15};
    int *result = twoSum(array, 4, 7+15, &size);
    for (int i = 0; i < size; i++) {
        printf("-- %d \n", result[i]);
    }
    
    return 0;
}

File: AlgorithmDataStructure/LeetCode/1-TwoSum/sum.swift


class Solution {
    
    func twoSumIndex(_ nums: [Int], _ target: Int) -> [Int] {
        var dict = [Int: Int]()
        for (i, num) in nums.enumerated() {
            if let lastIndex = dict[target - num] {
                return [lastIndex, i]
            } else {
                dict[num] = i
            }
        }
        fatalError("no")
    }
    
    func twoSum(nums: [Int], _ target: Int) -> Bool {
        var set = Set<Int>()
        for num in nums {
            if set.contains(target - num) {
                return true
            }
            set.insert(num)
        }
        return false
    }

    
    //let array = [1,3,4,5,6,7,8,9]
    //twoSum(nums: array, 10) //true
    //twoSumIndex(nums: array, target: 10) // [2,4]

}
